tests/ui_tools/test_boxes.py

```diff
@@ -124,3 +124,23 @@ def test_autocomplete_streams(self, write_box, streams_fixture,
             streams_fixture[len(streams_fixture)//2:]]
         typeahead_string = write_box.autocomplete_streams(text, state)
         assert typeahead_string == required_typeahead
+
+    @pytest.mark.parametrize('text, state, required_typeahead', [
+        (':rock_o', 0, ':rock_on:'),
+        (':rock_o', 1, ':rock_o'),
+        (':smi', 0, ':smiley:'),
+        (':smi', 1, ':smirk:'),
+        (':smi', 2, ':smile:'),
```

Comment:
This was what I meant by the order of cycling - it's not too bad as it is, but my expectation was that it would be in alphabetical order.

Reply:
If we want to keep it alphabetical, we could perhaps sort the entire list of emojis after retrieving it. This would be a one time thing, so it shouldn't be much of a performance issue.